.ci/jenkins/scripts/imageUtils.groovy

```diff
@@ -0,0 +1,66 @@
+containerEngine = 'docker'
+containerTlsOptions = ''
+
+void loginRegistry() {
+    loginContainerRegistry(getOperatorImageRegistry(), getOperatorImageRegistryCredentials())
+}
+
+void loginContainerRegistry(String registry, String credsId) {
+    echo "Using credentials ($credsId) to login to registry ($registry)"
+    withCredentials([usernamePassword(credentialsId: credsId, usernameVariable: 'REGISTRY_USER', passwordVariable: 'REGISTRY_PWD')]) {
+        sh "${containerEngine} login ${containerTlsOptions} -u ${REGISTRY_USER} -p ${REGISTRY_PWD} ${registry}"
+    }
+}
+
+String getOperatorImageRegistryCredentials() {
+    return params.OPERATOR_IMAGE_REGISTRY_CREDENTIALS
+}
+
+String getOperatorImageRegistry() {
+    return params.OPERATOR_IMAGE_REGISTRY
+}
+
+String getImageFullName(String namespace, String image, String tag) {
+    return "${namespace}/${image}:$tag"
+}
+
+String getImageFullNameWithRegistry(String registry, String namespace, String image, String tag) {
+    return "${registry}/${getImageFullName(namespace, image, tag)}"
+}
+
+void pullImage(String image) {
+    retry(env.MAX_REGISTRY_RETRIES ?: 1) {
+        sh "${containerEngine} pull ${containerTlsOptions} ${image}"
+    }
+}
+
+void tagImage(String oldImage, String newImage) {
+    sh "${containerEngine} tag ${oldImage} ${newImage}"
+}
+
+void pushImage(String image) {
+    loginRegistry()
+    retry(env.MAX_REGISTRY_RETRIES ?: 1) {
+        sh "${containerEngine} push ${containerTlsOptions} ${image}"
+    }
+}
+
+boolean removeQuayTag(String namespace, String imageName, String tag) {
+    String image = "quay.io/${namespace}/${imageName}:${tag}"
+    echo "Removing a temporary image tag ${image}"
+    try {
+        def output = 'false'
+       // withCredentials([usernamePassword(credentialsId: getOperatorImageRegistryCredentials(), usernameVariable: 'QUAY_USER', passwordVariable: 'QUAY_TOKEN')]) {
+        withCredentials([string(credentialsId: 'rsynek_registry_oauth_token', variable: 'QUAY_TOKEN')]) {
```

Comment:
TODO: switch to a token defined by the calling job.

What kind of credentials do we need here?

Reply:
I assume it's not just a username and password of an account with sufficient permissions; it should be a token created in the "Applications" menu of the organization settings (let's discuss this in a call).